Feature/creating second index on enrollment

Jordan-Ng opened the pull request:
- renamed decorator file to dependency_injection.py, dependency injection looks cleaner
- adding return statements to delete_item on Dynamo wrapper class
- tweaking models in models.py to be consistent with db structure
- adding minor details to endpoints + explanation for each in registrar_router.py
- tweaking seed.py to be consistent with required structure
- added notes and example usage for dependencies in student route (student_router.py)
- tweaked registrar endpoints + student's get_available_classes endpoint to support dependency injection functions"

Jordan-Ng committed:
WIP - preliminary codes for registrar routes
- added command to stop redis server before running
- added dynamo constant table mapping for dynamo wrapper
- changes to user_decorator to use dependency injection instead
- added new models to models.py
- added helper functions for registrar routes
- update seed.py to be consistent with requirements
- commented out decorator logic on student route
- add propogate claim functionalities for registrar routes for krakend.json

Jordan-Ng committed:
registrar endpoints
- renamed decorator file to dependency_injection.py, dependency injection looks cleaner
- adding return statements to delete_item on Dynamo wrapper class
- tweaking models in models.py to be consistent with db structure
- adding minor details to endpoints + explanation for each in registrar_router.py
- tweaking seed.py to be consistent with required structure
- added notes and example usage for dependencies in student route (student_router.py)
- tweaked registrar endpoints + student's get_available_classes endpoint to support dependency injection functions

NLTN commented on etc/krakend.json:
```diff
@@ -55,7 +56,8 @@
           "extra_config": {
             "backend/http": {
-              "return_error_code": true
+              "return_error_code": true,
+              "return_error_details": "enrollment_service"
```
```suggestion
              "return_error_code": true
```
Because we want the Gateway to forward the Error codes from the Backends, we need to removed this line.

When you use return_error_details, all status codes returned to the client are 200.
Read more: https://www.krakend.io/docs/backends/detailed-errors/#return-backend-errors-in-a-new-key

NLTN commented on etc/krakend.json:
```diff
@@ -85,7 +94,8 @@
           "extra_config": {
             "backend/http": {
-              "return_error_code": true
+              "return_error_code": true,
+              "return_error_details": "enrollment_service"
```
```suggestion
              "return_error_code": true
```
Because we want the Gateway to forward the Error codes from the Backends, we need to removed this line.

When you use return_error_details, all status codes returned to the client are 200.
Read more: https://www.krakend.io/docs/backends/detailed-errors/#return-backend-errors-in-a-new-key

NLTN commented on etc/krakend.json:
```diff
@@ -115,7 +132,8 @@
           "extra_config": {
             "backend/http": {
-              "return_error_code": true
+              "return_error_code": true,
+              "return_error_details": "enrollment_service"
```
```suggestion
              "return_error_code": true
```
Because we want the Gateway to forward the Error codes from the Backends, we need to removed this line.

When you use return_error_details, all status codes returned to the client are 200.
Read more: https://www.krakend.io/docs/backends/detailed-errors/#return-backend-errors-in-a-new-key

NLTN commented on etc/krakend.json:
```diff
@@ -127,25 +145,33 @@
           "extra_config": {
             "backend/http": {
-              "return_error_code": true
+              "return_error_code": true,
+              "return_error_details": "enrollment_service"
```
```suggestion
              "return_error_code": true
```
Because we want the Gateway to forward the Error codes from the Backends, we need to removed this line.

When you use return_error_details, all status codes returned to the client are 200.
Read more: https://www.krakend.io/docs/backends/detailed-errors/#return-backend-errors-in-a-new-key

NLTN commented on etc/krakend.json:
```diff
@@ -157,25 +183,33 @@
           "extra_config": {
             "backend/http": {
-              "return_error_code": true
+              "return_error_code": true,
+              "return_error_details": "enrollment_service"
```
```suggestion
              "return_error_code": true
```
Because we want the Gateway to forward the Error codes from the Backends, we need to removed this line.

When you use return_error_details, all status codes returned to the client are 200.
Read more: https://www.krakend.io/docs/backends/detailed-errors/#return-backend-errors-in-a-new-key

Jordan-Ng committed:
removing return error detail
- removing return_error_detail on krakend.json

Jordan-Ng committed:
changing enrollment period status table structure
- tweaked seed.py to rework enrollment_period_status table
- tweaked models.py, Dynamo.py, registrar_router.py and registrar_helper.py to support rework

Jordan-Ng committed:
resolving pesky import error

Jordan-Ng committed:
refactoring + seeding config

NLTN reviewed (approved):
Well done
